dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java

```diff
@@ -50,613 +50,617 @@
-				}
-
-				f.setIdentifier(newIdentifier.getId());
-				save(f,  user,  respectFrontEndPermissions);
-			}
-			parent = f;
-		}
-		return parent;
-	}
-
-	@CloseDBIfOpened
-	public List<Map<String, Serializable>> DBSearch(Query query, User user, boolean respectFrontendRoles) throws ValidationException,
-			DotDataException {
-		Map<String, String> dbColToObjectAttribute = new HashMap<String, String>();
-
-		if (UtilMethods.isSet(query.getSelectAttributes())) {
-
-			if (!query.getSelectAttributes().contains("title")) {
-				query.getSelectAttributes().add("title" + " as " + QueryResult.CMIS_TITLE);
-			}
-		} else {
-			List<String> atts = new ArrayList<String>();
-			atts.add("*");
-			atts.add("title" + " as " + QueryResult.CMIS_TITLE);
-			query.setSelectAttributes(atts);
-		}
-
-		return QueryUtil.DBSearch(query, dbColToObjectAttribute, null, user, false, respectFrontendRoles);
-	}
-
-	/**
-	 * @deprecated Not implemented because it does not take a user
-	 */
-
-
-	@CloseDBIfOpened
-	public List<Folder> findFoldersByHost(Host host, User user, boolean respectFrontendRoles) throws DotHibernateException {
-		return folderFactory.findFoldersByHost(host);
-	}
-
-	@CloseDBIfOpened
-	public  List<Link> getLinks(final Folder parent, final boolean working,
-								final boolean deleted, final User user,
-								final boolean respectFrontEndPermissions)
-								throws DotStateException, DotDataException, DotSecurityException {
+    @CloseDBIfOpened
+    public Folder findFolderByPath(final String path, final Host host,
+                                   final User user, final boolean respectFrontEndPermissions) throws DotStateException,
+            DotDataException, DotSecurityException {
+
+        final Folder folder = folderFactory.findFolderByPath(path, host);
+
+        if (folder != null && InodeUtils.isSet(folder.getInode()) &&
+                !permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)) {
+
+            // SYSTEM_FOLDER means if the user has permissions to the host, then they can see host.com/
+            if(FolderAPI.SYSTEM_FOLDER.equals(folder.getInode())) {
+                if(!Host.SYSTEM_HOST.equals(host.getIdentifier())){
+                    if(!permissionAPI.doesUserHavePermission(host, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)) {
+                        throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to read folder " + folder.getPath());
+                    }
+                }
+            }
+
+        }
+
+        return folder;
+    }
+
+
+    public Folder findFolderByPath(String path, String hostid, User user, boolean respectFrontEndPermissions) throws DotStateException,
+            DotDataException, DotSecurityException {
+        if(user==null) user = APILocator.getUserAPI().getSystemUser();
```

Comment:
use two lines to do this, such as
```
if(user==null) { 
user = APILocator.getUserAPI().getSystemUser();
}
```

Eventually someone can introduce an error with the current code